be/src/exec/es/es_scroll_parser.cpp

```diff
@@ -111,6 +111,17 @@ static const std::string INVALID_NULL_VALUE =
         }                                                                    \
     } while (false)
 
+#define RETURN_ERROR_IF_COL_IS_NOT_ARRAY(col, type)                          \
+    do {                                                                     \
+        if (!col.IsArray()) {                                                \
```

Comment:
duplicate code with RETURN_ERROR_IF_COL_IS_ARRAY except '!' , so they can be combined into one with one more argument.